.github/workflows/build-and-test.yml

```diff
@@ -26,7 +26,7 @@ jobs:
       - name: Setup Foundry
         uses: foundry-rs/foundry-toolchain@v1
         with:
-          version: nightly-87bc53fc6c874bd4c92d97ed180b949e3a36d78c
```

Comment:
This was set on purpose because nightly foundry builds breaks the build CI quite often so we decided to use an exact commit. Either keep it untouched either set to the latest commit (if the build passes)

Reply:
revamped and passing at https://github.com/ubiquity/ubiquity-dollar/pull/704/commits/d89aaa73d7f3cd4831b256c942badee62353dacc